charts/asset-manager/templates/ingress.yaml

```
@@ -9,7 +9,7 @@ metadata:
     {{- include "asset-manager.labels" . | nindent 4 }}
     app.kubernetes.io/component: web
   annotations:
-    {{- toYaml .Values.ingress.annotations | nindent 4 }}
+    {{- (tpl (toYaml .Values.ingress.annotations) .) | trim | nindent 4 }}
```

Does the controller actually accept YAML, or was this only working because toYaml just happens to preserve JSON syntax? More robust/clearer to use `toJson` explicitly?
```suggestion
    {{- (tpl (toJson .Values.ingress.annotations) .) | trim | nindent 4 }}
```

---

Just tested and the controller doesn't accept YAML, so has to be annotation value for the condition has to be JSON.

Was using `toYaml` to preserve the YAML structure for the rest of the annotations.
If we use `toJson`, we get:
```yaml
  annotations:
    {"alb.ingress.kubernetes.io/conditions.release-name":"- field: host-header\n  hostHeaderConfig:\n    values:\n      - 'assets-eks.integration.publishing.service.gov.uk'\n","alb.ingress.kubernetes.io/group.name":"assets-origin","alb.ingress.kubernetes.io/group.order":"11","alb.ingress.kubernetes.io/healthcheck-path":"/healthcheck/ready","alb.ingress.kubernetes.io/listen-ports":"[{\"HTTP\": 80}, {\"HTTPS\": 443}]","alb.ingress.kubernetes.io/load-balancer-attributes":"access_logs.s3.enabled=true,access_logs.s3.bucket=govuk-integration-aws-logging,access_logs.s3.prefix=elb/assets-origin-eks","alb.ingress.kubernetes.io/load-balancer-name":"assets-origin","alb.ingress.kubernetes.io/scheme":"internet-facing","alb.ingress.kubernetes.io/ssl-redirect":"443","alb.ingress.kubernetes.io/target-type":"ip"}
```
Which then get reformatted back to YAML when applied to the cluster anyways, and the annotation value is actually a string so stays as YAML (which doesn't work).

---

Oops, yes of course - I didn't consider that it's on the whole set of annotations and not just the new one. My suggestion doesn't make sense, sorry.